### ISSUE TITLE
issue with angular and html5mode set as true.

### ISSUE THREAD

**Issue by lili21** [nwjs/nw.js#5478]
[sample app to reproduce the issue](https://github.com/lili21/nwjs-issue)

Play with the sample app, and You will find out the iframe content looks like "load twice". 

If you like to dig in, open the DevTools, and check out the iframe actual url, which will be `http://localhost:8283/#/app/iframecontent` instead of `http://localhost:8283/app/iframecontent`.

You can review #5460 for more details.

**Comment by ghostoy** [comment]
This issue is caused by Angular.js incorrectly detect NW.js app as Chrome App, and prevent using html5Mode. Currently you can workaround it by patching following lines in `$SnifferProvider`:

```js
function $SnifferProvider() {
  this.$get = ['$window', '$document', function($window, $document) {
    var eventSupport = {},
        isChromePackagedApp = // ...
        isNwjs = $window.nw && $window.nw.process,
        hasHistoryPushState = (isNwjs || !isChromePackagedApp) && $window.history && $window.history.pushState,
// ...
```

I've submitted [an issue for Angular.js](angular/angular.js/issues/15474). Please follow up the issue for updates.

**Comment by lili21** [comment]
issue link is broken.

**Comment by lili21** [comment]
link is broken. But thanks.

**Comment by ghostoy** [comment]
The upstream bug was fixed in master and 1.6.x branch. See https://github.com/angular/angular.js/issues/15474
